### PR TITLE
Re-import samples for packaging

### PR DESCRIPTION
**Pull request by bbowman**

### CODE REVIEW

File: samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt [hunk 1]
```diff
@@ -87,15 +87,15 @@ PRODUCT_BUNDLE_IDENTIFIER = com.Microsoft.CoreAnimationTest
 PRODUCT_NAME = CoreAnimationTest
 PRODUCT_TYPE = com.apple.product-type.application
 PROJECT = CoreAnimationTest
-PROJECT_DIR = E:/GIT_REPOS/winobjc/samples/CoreAnimationTest
-PROJECT_FILE_PATH = E:\GIT_REPOS\winobjc\samples\CoreAnimationTest\CoreAnimationTest.xcodeproj
+PROJECT_DIR = Y:/Arcadia/WinObjC/samples/CoreAnimationTest
```
Comment (pradipd) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt]: >Arcadia [](start = 17, length = 7)

What is arcadia? :)

File: samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest.vcxproj [hunk 2]
```diff
@@ -89,7 +89,7 @@
       <HeaderMap>Project</HeaderMap>
       <IncludePaths>$(SolutionPublicHeadersDir)</IncludePaths>
       <ObjectiveCARC>true</ObjectiveCARC>
-      <ObjectiveCModules>false</ObjectiveCModules>
```
Comment (pradipd) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest.vcxproj]: >false [](start = 25, length = 5)

Change back to false.  
This was done to speed up the builds.

**Comment by pradipd** [comment]
:shipit:

File: samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest.vcxproj [hunk 3]
```diff
@@ -113,7 +113,7 @@
...
+      <ObjectiveCModules>true</ObjectiveCModules>
```
Comment (memontic-ms) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest.vcxproj]: >true [](start = 25, length = 4)

It builds okay like this? Alright then.

File: samples/GLKitComplex/GLKitComplex.vsimporter/GLKitComplex-WinStore10/GLKitComplex.vcxproj [hunk 4]
```diff
@@ -59,6 +58,7 @@
     <TargetName>GLKitComplex</TargetName>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="ExtensionSettings" />
```
Comment (memontic-ms) [inline samples/GLKitComplex/GLKitComplex.vsimporter/GLKitComplex-WinStore10/GLKitComplex.vcxproj]: ><ImportGroup Label="ExtensionSettings" /> [](start = 2, length = 41)

I should really go look if this is EVERY needed by vsimporter...

**Comment by memontic-ms** [comment]
>     </Text>

Did you actually open up this solution in VS and save it? I'm curious if vsimporter should have added the project.json to the filters file or if VS has been doing it automatically for me and I haven't noticed. +1 issue for me.

---
Refers to: samples/HelloGLKit/HelloGLKit.vsimporter/HelloGLKit-WinStore10/HelloGLKit.vcxproj.filters:72 in c66e169. [](commit_id = c66e169ebc9a3cbc4694b7bba2a2519b7dc7f7a0, deletion_comment = False)

File: samples/KVOPerf/KVOPerf.vsimporter/KVOPerf-WinStore10/KVOPerf.vcxproj [hunk 5]
```diff
@@ -189,8 +203,7 @@
     <Xml Include="default.rd.xml" />
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
-
-  <PropertyGroup Label="Overrides">
-    <StarboardDefaultLibs>CoreGraphics.lib;CoreText.lib;Foundation.lib;QuartzCore.lib;Starboard.lib;UIKit.lib;libdispatch.lib</StarboardDefaultLibs>
-  </PropertyGroup>
```
Comment (memontic-ms) [inline samples/KVOPerf/KVOPerf.vsimporter/KVOPerf-WinStore10/KVOPerf.vcxproj]: Interesting

File: samples/WinRTSample/WinRTSample.vsimporter/WinRTSample-WinStore10/WinRTSample.vcxproj [hunk 6]
```diff
@@ -201,5 +215,7 @@
   </ItemGroup>
   <ItemGroup />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
-
+  <ItemGroup>
+    <None Include="project.json" />
+  </ItemGroup>
```
Comment (memontic-ms) [inline samples/WinRTSample/WinRTSample.vsimporter/WinRTSample-WinStore10/WinRTSample.vcxproj]: Vsimporter should have grouped this in the item group with everything else. Don't know what went on here.

File: samples/WinRTSample/WinRTSample.vsimporter/WinRTSample-WinStore10/WinRTSample.vcxproj [hunk 7]
```diff
@@ -201,5 +215,7 @@
   </ItemGroup>
   <ItemGroup />
```
Comment (memontic-ms) [inline samples/WinRTSample/WinRTSample.vsimporter/WinRTSample-WinStore10/WinRTSample.vcxproj]: ><ItemGroup /> [](start = 2, length = 13)

Vsimporter why did you do this. +1 for me to investigate

File: samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/Package.appxmanifest [hunk 8]
```diff
@@ -45,6 +45,5 @@
 
   <Capabilities>
     <Capability Name="internetClient" />
-	<DeviceCapability Name="location" />
```
Comment (memontic-ms) [inline samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/Package.appxmanifest]: ><DeviceCapability Name="location" /> [](start = 1, length = 36)

Probably need this. Did you check out WOCCatalog's location features to see if they work... correctly?

File: samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj [hunk 9]
```diff
@@ -88,12 +89,13 @@
...
       <IncludePaths>$(SolutionPublicHeadersDir)</IncludePaths>
       <ObjectiveCARC>true</ObjectiveCARC>
-      <ObjectiveCModules>false</ObjectiveCModules>
+      <ObjectiveCModules>true</ObjectiveCModules>
       <OptimizationLevel>Disabled</OptimizationLevel>
       <PreprocessorDefinitions>DEBUG=1</PreprocessorDefinitions>
     </ClangCompile>
     <Link>
       <GenerateDebugInformation>true</GenerateDebugInformation>
+      <AdditionalDependencies>%(AdditionalDependencies);Accelerate.lib;AudioToolbox.lib;CoreImage.lib;CoreMotion.lib;GLKit.lib;OpenGLES.lib</AdditionalDependencies>
```
Comment (memontic-ms) [inline samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj]: ><AdditionalDependencies>%(AdditionalDependencies);Accelerate.lib;AudioToolbox.lib;CoreImage.lib;CoreMotion.lib;GLKit.lib;OpenGLES.lib</AdditionalDependencies> [](start = 6, length = 158)

Interesting that these are newly added. Know why?

File: samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj [hunk 10]
```diff
@@ -88,12 +89,13 @@
       <HeaderMap>Project</HeaderMap>
       <IncludePaths>$(SolutionPublicHeadersDir)</IncludePaths>
```
Comment (memontic-ms) [inline samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj]: ><IncludePaths>$(SolutionPublicHeadersDir)</IncludePaths> [](start = 6, length = 56)

Might want the newer vsimporter version where IncludePaths don't get stomped.

File: tests/testapps/AddressBookSample/AddressBookSample.vsimporter/AddressBookSample-WinStore10/Package.appxmanifest [hunk 11]
```diff
@@ -1,29 +1,49 @@
...
+        BackgroundColor="transparent">
+        <uap:DefaultTile Wide310x150Logo="Assets\Wide310x150Logo.png"/>
         <uap:SplashScreen Image="Assets\SplashScreen.png" />
       </uap:VisualElements>
     </Application>
   </Applications>
+
   <Capabilities>
     <Capability Name="internetClient" />
-    <uap:Capability Name="contacts" />
```
Comment (memontic-ms) [inline tests/testapps/AddressBookSample/AddressBookSample.vsimporter/AddressBookSample-WinStore10/Package.appxmanifest]: ><uap:Capability Name="contacts" /> [](start = 4, length = 34)

This is probably necessary for an "AddressBook"

Comment (memontic-ms) [inline tests/testapps/AddressBookSample/AddressBookSample.vsimporter/AddressBookSample-WinStore10/Package.appxmanifest]: We should really be better about automatically adding these capabilities. +1 to me.

---
In reply to: [102806885](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102806885) [](ancestors = 102806885)

**Comment by memontic-ms** [comment]
> <?xml version="1.0" encoding="utf-8"?>

This part is definitely in the templates. I don't know how it's always getting nixed. Is this a VS 2015 thing?

---
Refers to: tests/testapps/CGCatalog/CGCatalog.vsimporter/CGCatalog-WinStore10/CGCatalog.vcxproj:1 in c66e169. [](commit_id = c66e169ebc9a3cbc4694b7bba2a2519b7dc7f7a0, deletion_comment = True)

File: samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt [hunk 12]
```diff
@@ -87,15 +87,15 @@ PRODUCT_BUNDLE_IDENTIFIER = com.Microsoft.CoreAnimationTest
...
+PROJECT_FILE_PATH = Y:\Arcadia\WinObjC\samples\CoreAnimationTest\CoreAnimationTest.xcodeproj
```
Comment (memontic-ms) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt]: @asimonov-msft Why are PROJECT_DIR, PROJECT_FILE_PATH, SOURCE_ROOT, and SRCROOT even needed in this file? Can we get rid of them altogether?

File: samples/WinRTSample/WinRTSample.vsimporter/WinRTSample-WinStore10/WinRTSample.vcxproj (hunk 6 above)
Comment (bbowman) [inline samples/WinRTSample/WinRTSample.vsimporter/WinRTSample-WinStore10/WinRTSample.vcxproj]: WinRTSample and KVO Perf didn't have the original xcode around so I manually did them.

---
In reply to: [102804684](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102804684) [](ancestors = 102804684)

File: samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj (hunk 10 above)
Comment (bbowman) [inline samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj]: yeah probably but this is fine for now. Just trying to get builds working. We can re-re-import later if we want.

---
In reply to: [102805999](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102805999) [](ancestors = 102805999)

File: samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj (hunk 9 above)
Comment (bbowman) [inline samples/WOCCatalog/WOCCatalog.vsimporter/WOCCatalog-WinStore10/WOCCatalog.vcxproj]: no clue!

---
In reply to: [102805890](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102805890) [](ancestors = 102805890)

File: samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt (hunk 12 above)
Comment (bbowman) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt]: that would be awesome. Getting rid of as many of these as possible / converting them into msbuild props would make me ever so happy.

---
In reply to: [102809102](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102809102) [](ancestors = 102809102)

**Comment by bbowman** [comment]
>     </Text>

Yes. I opened them all in VS to build. I didn't notice if it overwrote these filters from what vsimporter did. +1 on you checking that out.

---
In reply to: [282103326](https://github.com/Microsoft/WinObjC/pull/2056#issuecomment-282103326) [](ancestors = 282103326)

---
Refers to: samples/HelloGLKit/HelloGLKit.vsimporter/HelloGLKit-WinStore10/HelloGLKit.vcxproj.filters:72 in c66e169. [](commit_id = c66e169ebc9a3cbc4694b7bba2a2519b7dc7f7a0, deletion_comment = False)

**Comment by bbowman** [comment]
> <?xml version="1.0" encoding="utf-8"?>

Yeah not sure. Wouldn't surprise me if it was vs2017 rewriting it.

---
In reply to: [282107952](https://github.com/Microsoft/WinObjC/pull/2056#issuecomment-282107952) [](ancestors = 282107952)

---
Refers to: tests/testapps/CGCatalog/CGCatalog.vsimporter/CGCatalog-WinStore10/CGCatalog.vcxproj:1 in c66e169. [](commit_id = c66e169ebc9a3cbc4694b7bba2a2519b7dc7f7a0, deletion_comment = True)

**Comment by memontic-ms** [comment]
:shipit:

**Comment by bbowman** [comment]
> <?xml version="1.0" encoding="utf-8"?>

going to let it be for now though.

---
In reply to: [282132028](https://github.com/Microsoft/WinObjC/pull/2056#issuecomment-282132028) [](ancestors = 282132028,282107952)

---
Refers to: tests/testapps/CGCatalog/CGCatalog.vsimporter/CGCatalog-WinStore10/CGCatalog.vcxproj:1 in c66e169. [](commit_id = c66e169ebc9a3cbc4694b7bba2a2519b7dc7f7a0, deletion_comment = True)

File: samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt (hunk 12 above)
Comment (pradipd) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt]: I thought this file was never used.  My *limited* understanding is that it was just output.  Could be wrong.

---
In reply to: [102826848](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102826848) [](ancestors = 102826848,102809102)

Comment (pradipd) [inline samples/CoreAnimationTest/CoreAnimationTest.vsimporter/CoreAnimationTest-WinStore10/CoreAnimationTest-Debug-xcvars.txt]: output /debug info.  As in these were the original xcode variables.  

---
In reply to: [102846565](https://github.com/Microsoft/WinObjC/pull/2056#discussion_r102846565) [](ancestors = 102846565,102826848,102809102)

**Comment by DHowett-MSFT** [comment]
@pradipd the xcvars are used for Info.plist parsing. The user can have multiple info.plist files for different configurations, and different configurations can have different variable maps, and info.plist supports interpolation of those variables.

**Pull request merged**